src/exoscale/vinyl/store.clj

```diff
@@ -302,3 +384,52 @@
    all records for a specific composite key prefix"
   [txn-context record-type items]
   (delete-by-range txn-context (all-of-range txn-context record-type items)))
+
+(defn long-query-reducer
+  "A reducer over large ranges. Accepts queries as per `execute-query`. Results
+   are reduced into an accumulator with the help of the reducing function `f`.
+   The accumulator is initiated to `init`. `clojure.core.reduced` is honored.
+
+   Obviously, this approach does away with any consistency guarantees usually
+   offered by FDB.
+
+   Results being accumulated in memory, this also means that care must be
+   taken with the accumulator."
+  ;; Some cliff notes to read the code below.
+  ;; The basic idea is that scanning is done while it works, up until
+  ;; the point where an exception will be raised.
+  ;;
+  ;; By default, `run-async` uses a simple exponential back-off algorithm
+  ;; between transaction function retries. In this case we want to avoid that.
+  ;; To that effect, a runner is created with specific parameters (a large
+  ;; `max-attempts` value, as well as minimum viable delays.
+  ;;
+  ;; We then call `apply-reduce` on the returned cursor from our query with a
+  ;; twist: every visited element will get its continuation stored in an atom.
+  ;; When interrupted, the function will be retried, which pops the last seen
+  ;; continuation.
+  ;;
+  ([db f init query]
+   (long-query-reducer db f init query {}))
```

Comment:
In apply-transforms we use the keys `reducer` and `reducer-init`.
Maybe we could use the same names (at least for `f`) ?

https://github.com/exoscale/vinyl/blob/master/src/exoscale/vinyl/cursor.clj#L38-L39

Reply:
I wanted to match the args to `reduce` (but seems I missed the target since they have `val`, not `init`). I will adapt to match `reduce` completely